src/producer.rs

```diff
@@ -370,14 +372,18 @@ pub fn to_substrait_agg_measure(
     }
 }
 
-fn to_substrait_jointype(join_type: JoinType) -> join_rel::JoinType {
+fn to_substrait_jointype(join_type: JoinType) -> Result<join_rel::JoinType> {
     match join_type {
-        JoinType::Inner => join_rel::JoinType::Inner,
-        JoinType::Left => join_rel::JoinType::Left,
-        JoinType::Right => join_rel::JoinType::Right,
-        JoinType::Full => join_rel::JoinType::Outer,
-        JoinType::Anti => join_rel::JoinType::Anti,
-        JoinType::Semi => join_rel::JoinType::Semi,
+        JoinType::Inner => Ok(join_rel::JoinType::Inner),
+        JoinType::Left => Ok(join_rel::JoinType::Left),
+        JoinType::Right => Ok(join_rel::JoinType::Right),
+        JoinType::Full => Ok(join_rel::JoinType::Outer),
+        JoinType::LeftAnti => Ok(join_rel::JoinType::Anti),
+        JoinType::LeftSemi => Ok(join_rel::JoinType::Semi),
+        _ => Err(DataFusionError::NotImplemented(format!(
```

Comment:
DataFusion now supports `RightAnti` and `RightSemi` and I do not know if there is an equivalent in Substrait

Reply:
It doesn't look like it. If we really want to support it, one option is to switch the left and right inputs. I think this would work functionally but roundtrip tests will not work. I also checked the Substrait issues but I don't see any mention of right semi/anti. Maybe we can also submit a feature request if necessary?